src/libponyc/pass/syntax.c

```diff
@@ -1126,10 +1126,12 @@ ast_result_t pass_syntax(ast_t** astp, pass_opt_t* options)
     case TK_VALUEFORMALARG:
     case TK_VALUEFORMALPARAM:
       ast_error(options->check.errors, ast,
-        "Value formal parameters not yet supported");
+        "Value formal parameters not yet supported. "
+        "Note that many functions including array indexing use the apply "
```

Comment:
I'd suggest using `ast_error_continue` here to get the `apply` message in the `Info` part of the error report.